refactor: 정규식들 파일 레벨 정의 방식으로 변경

easyhooon opened the pull request:
<!--
✅ PR 제목 작성 가이드
형식: <라벨>: <작업 요약>
예: feat: 로그인 페이지 구현, fix: 버튼 클릭 버그 수정
-->

## 🔗 관련 이슈
<!-- 이 PR과 연결된 이슈 번호를 명시해주세요 (예: Close #123) -->
- Close #

## 📙 작업 설명
<!-- 주요 수정 사항이나 개발 내용을 요약해주세요 -->
- ~~derivedStateOf 블럭내에 계산이 진행될때마다 생성 비용이 큰 Regex 객체가 매번 생성되는 것을 remember로 캐싱했습니다.~~
- 기존의 개선 방식은 WaitingDialog 컴포저블이 새로 생성될때마다 어쨌든 Regex 객체를 생성하므로 비효율적.
- object로 관리하여, 정규식 첫 사용시 메모리에 로드된 후 이후 다시 초기화가 되지 않도록  변경(싱글톤)
- 누락된 versionCode update(47 -> 48)

## 🧪 테스트 내역 (선택)
- [x] 주요 기능 정상 동작 확인
- [x] 브라우저/기기에서 동작 확인
- [x] 엣지 케이스 테스트 완료
- [x] 기존 기능 영향 없음

## 📸 스크린샷 또는 시연 영상 (선택)
<!-- UI 변경사항이 있다면 이미지나 GIF를 첨부해주세요 -->

## 💬 추가 설명 or 리뷰 포인트 (선택)
<!-- 리뷰어가 중점적으로 봐야 할 부분이나 설명이 필요한 내용을 자유롭게 작성해주세요 -->
-


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- 성능 개선
  - 대기 다이얼로그의 입력 검증이 통일되어 안정성과 응답성이 소폭 향상되었습니다.
- Bug Fixes
  - 대기 신청 시 허용되는 인원 범위가 1~10으로 명확히 제한되어 잘못된 입력이 줄어듭니다.
  - 전화번호 검증 로직이 표준화되어 입력 처리 일관성이 향상되었습니다.
- Chores
  - 앱 내부 버전이 48로 업데이트되어 배포 준비가 완료되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

easyhooon committed:
[refactor] 전화번호 관련 정규식을 remember 로 캐싱

easyhooon committed:
[chore] versionCode update

47 -> 48

ikseong00 reviewed (approved):
확인했습니다!

말씀해주신 버전 코드가 버전명인줄 알았습니다.. 감사합니다!! 👍

jisungbin commented on core/ui/src/main/kotlin/com/unifest/android/core/ui/component/WaitingDialog.kt:
```diff
@@ -203,10 +203,12 @@ fun WaitingDialog(
     onPrivacyPolicyClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
+    val phoneNumberRegex = remember { Regex("^01[0-9]\\d{8}\$") }
```
이건 상수로 쓰이는 정규식이니 파일 레벨 정의로 옮기는 건 어떨까요?
(이런 상수 정규식들을 모아둔 자바 기본 클래스가 있었던 거 같은데 기억이 안나네요 ㅠ)

easyhooon replied:
@jisungbin 맞슴다 사실 이렇게 처리해도 다이얼로그가 뜰때마다 새로 객체를 생성하긴 힐거라, 더 위로 올려야겠네요!

easyhooon replied:
적용 완료 하였습니다! [1f477d3](https://github.com/Project-Unifest/unifest-android/pull/442/commits/1f477d3da7d8012a2a59bac0da55efb25157f049)

jisungbin replied:
감사합니당 ><

easyhooon committed:
[refactor] 정규식들 파일 레벨 정의 방식으로 변경

기존 방식은 WaitingDialog 컴포저블이 새로 생성될때마다 Regex 객체를 생성하므로 비효율적.

object로 관리하여, 정규식 첫 사용시 메모리에 올려 이후 다시 초기화가 되지 않도록  변경(싱글톤)
